### DISCUSSION TITLE
How do I edit the corresponding Database User with new, user-provided data?

### DISCUSSION THREAD

**Discussion by MeowcaTheoRange** [nextauthjs/next-auth#7678] [Help]
Here's a scenario: You, as a user, have just signed up for **COOL NEW SERVICE LTD**.
**COOL NEW SERVICE LTD** has a signup page after you authorize with your provider. It lets you choose a username, user color let's say, a bio, and maybe more.
Then, after you enter all of the information, it will update your profile with the data that you gave.
How do I do this with Auth.js? I'm using Database Sessions (MongoDB adapter) and Next.js App Router. I simply need a way to 
1. find out what user document to modify,
2. if the client session is allowed to modify that document.

When I use `useSession` on the client, all I get is a barebones copy of the database user, which does not help me identify which MongoDB document the client is.
I'll have to find out what account and/or session the client is, so I can match it against *those* MongoDB documents, and then actually get the user document.

Basically:
![image](https://github.com/nextauthjs/next-auth/assets/58280776/330a8e5c-2485-43d1-a884-bbdaba744a5b)
